pkg/vm/vm_test.go

```diff
@@ -3032,6 +3032,116 @@ func TestHASH256(t *testing.T) {
 	assert.Equal(t, res, hex.EncodeToString(vm.estack.Pop().Bytes()))
 }
 
+func TestAnd(t *testing.T) {
+	prog := makeProgram(opcode.AND)
+	vm := load(prog)
+	vm.estack.PushVal(1)
+	vm.estack.PushVal(1)
+	runVM(t, vm)
+	assert.Equal(t, int64(1), vm.estack.Pop().BigInt().Int64())
+}
+
+func TestOr(t *testing.T) {
+	prog := makeProgram(opcode.OR)
+	vm := load(prog)
+	vm.estack.PushVal(1)
+	vm.estack.PushVal(0)
+	runVM(t, vm)
+	assert.Equal(t, int64(1), vm.estack.Pop().BigInt().Int64())
+}
+
+func TestXOr(t *testing.T) {
+	prog := makeProgram(opcode.XOR)
+	vm := load(prog)
+	vm.estack.PushVal(1)
+	vm.estack.PushVal(0)
+	runVM(t, vm)
+	assert.Equal(t, int64(1), vm.estack.Pop().BigInt().Int64())
```

Comment:
Given this set of tests I can implement OR as XOR or XOR as OR and nobody would notice. We need stronger tests covering different expected input-output variations (not just here, but for also for other opcodes).

Reply:
Added different test-cases and re-organized tests structure.